starter-code/src/functions-and-arrays.js

```diff
@@ -74,8 +192,43 @@ var wordsCount = [
   'disobedience',
   'matter'
 ];
+
+
+function howManyTimes (array, word) {
+
+  var counter = 0;
+
+  if (array.length === 0){
+    return false;
+  }
+  else {
+    for (var i=0; i< array.length; i++) {
+      if (array[i] === word) {
+        counter ++;
+      }
+    }
+  }
+return counter;
+}
 // Bonus Quest
 
+function countUp (mat, x, y){
+  count = 0;
+  for (var i=0; i<4; i++) {
+    count *= mat[i][y];
+  }  
+  return count;
+}
+
+
+function greatestProduct (array) {
+  var temp =0;
+
+    
+
+  return temp;
+}
+
 var matrix = [
   [8, 2, 22, 97, 38, 15, 0, 40, 0, 75, 4, 5, 7, 78, 52, 12, 50, 77, 91, 8],
   [49, 49, 99, 40, 17, 81, 18, 57, 60, 87, 17, 40, 98, 43, 69, 48, 4, 56, 62, 0],
```

Comment:
Here is the correction for the Matrix: 

```
function greatestProduct(m) {
  var horizontal, vertical, diagonal;
  var greatProduct = 0;

  for (var i=0, nRows = m.length; i < nRows - 1; i++) {
    for (var j=0, nCols = m[0].length; j < nCols - 1; j++) {
      // 4 horizontal elements
      if (j < nCols-3) {
        horizontal = m[i][j] * m[i][j+1]   * m[i][j+2]   * m[i][j+3];
      }

      // 4 vertical elements
      if (i < nRows-3) {
        vertical   = m[i][j] * m[i+1][j]   * m[i+2][j]   * m[i+3][j];
      }

      // 4 Diagonal elements
      if (i < nRows-3 && j < nCols-3) {
        diagonal   = m[i][j] * m[i+1][j+1] * m[i+2][j+2] * m[i+3][j+3];
      }

      greatProduct = Math.max(greatProduct, horizontal, vertical, diagonal);
    }
  }

  return greatProduct;
}
```

Reply:
It's globally well done! The code is clean and not too long. 💪

Keep up the good work!